### PR TITLE
Task/rewording

### PR DESCRIPTION
**Pull request by simonamaggio**
- global error to total error fraction
- mpp to error tree

### CODE REVIEW

**Comment by AgatheG** [comment]
Do we want Error tree or Error analyzer tree?

File: README.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ error_analyzer = ErrorAnalyzer(model, feature_names=feature_names)
 error_analyzer.fit(X_test, y_test)
 
 # print metrics regarding the Model Performance Predictor
-print(error_analyzer.mpp_summary(X_test, y_test, output_dict=False))
+print(error_analyzer.evaluate(X_test, y_test, output_format='text'))
```
Comment (AgatheG) [inline README.md:76]: You should change `evaluate` to return the string (for now it prints it).
Also nitpick, we do not need not have `output_format='text'` as it is the default (same on l.136)

File: README.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ error_analyzer = ErrorAnalyzer(model, feature_names=feature_names)
...
 
 # plot the Model Performance Predictor Decision Tree
```
Comment (AgatheG) [inline README.md:78]:
```suggestion
# plot the Error Tree
```

File: README.md [hunk 3]
```diff
@@ -133,7 +133,7 @@ error_analyzer = ErrorAnalyzer(pipeline_model, feature_names=feature_names)
...
```
Comment (AgatheG) [inline README.md:138]:
```suggestion
# plot the Error Tree
```

File: examples/plot_mealy.py [hunk 4]
```diff
@@ -58,18 +58,18 @@
...
 
 ##############################################################################
-# Print metrics regarding the Model Performance Predictor.
+# Print metrics regarding the Error Tree.
 
 print(error_analyzer.evaluate(X_test, y_test, output_format='text'))
 
 ##############################################################################
-# Plot the Model Performance Predictor Decision Tree.
+# Plot the Error Decision Tree.
```
Comment (AgatheG) [inline examples/plot_mealy.py:72]: [Extreme nitpick]
Can be simply `Error Tree` as we use this term everywhere else? Not very important, for consistency

File: examples/plot_mealy_pipeline.py [hunk 5]
```diff
@@ -103,18 +103,18 @@
...
 
 ##############################################################################
-# Print metrics regarding the Model Performance Predictor.
+# Print metrics regarding the Error Tree.
 
 print(error_analyzer.evaluate(X_test, y_test, output_format='text'))
 
 ##############################################################################
-# Plot the Model Performance Predictor Decision Tree.
+# Plot the Error Decision Tree.
```
Comment (AgatheG) [inline examples/plot_mealy_pipeline.py:117]: Same comment here

File: examples/plot_mealy_pipeline.py [hunk 6]
```diff
@@ -144,20 +144,20 @@
...
-# The MPP might not be representative of the behavior of the primary model as the true primary
-# accuracy and the one estimated by the MPP are not close.
+# The Error Tree might not be representative of the behavior of the primary model as the true primary
+# accuracy and the one estimated by the Error Tree are not close.
 #
 # Model Failures
 # ^^^^^^^^^^^^^^
 #
-# Let's focus on the nodes of the MPP DecisionTree, in particular the leaf nodes
+# Let's focus on the nodes of the Error Decision Tree, in particular the leaf nodes
```
Comment (AgatheG) [inline examples/plot_mealy_pipeline.py:160]: Likewise

File: mealy/error_tree.py [hunk 7]
```diff
@@ -84,4 +84,4 @@ def _compute_ranking_arrays(self, n_purity_levels=ErrorAnalyzerConstants.NUMBER_
 
         #y = self._error_train_y
         n_total_errors = np.sum(wrongly_predicted_samples) #y[y == ErrorAnalyzerConstants.WRONG_PREDICTION].shape[0]
-        self._global_error = wrongly_predicted_samples.astype(float) / n_total_errors
\ No newline at end of file
+        self._total_error_fraction = wrongly_predicted_samples.astype(float) / n_total_errors
```
Comment (AgatheG) [inline mealy/error_tree.py:87]: Easier to cast one iinteger to float than a whole array
```suggestion
        self._total_error_fraction = wrongly_predicted_samples / float(n_total_errors)

```

File: mealy/error_tree.py [hunk 8]
```diff
@@ -84,4 +84,4 @@ def _compute_ranking_arrays(self, n_purity_levels=ErrorAnalyzerConstants.NUMBER_
 
         #y = self._error_train_y
         n_total_errors = np.sum(wrongly_predicted_samples) #y[y == ErrorAnalyzerConstants.WRONG_PREDICTION].shape[0]
```
Comment (AgatheG) [inline mealy/error_tree.py:86]:
```suggestion
        n_total_errors = np.sum(wrongly_predicted_samples)
```

File: mealy/tests/test_error_analyzer.py [hunk 9]
```diff
@@ -86,32 +86,32 @@ def test_with_scikit_pipeline():
...
         'primary_model_true_accuracy': primary_model_true_accuracy,
         'fidelity': fidelity
     }
 
     metric_reference = {
-        'mpp_accuracy_score': 0.8952863503761708,
-        'mpp_balanced_accuracy': 0.7209989546416461,
+        'error_tree_accuracy_score': 0.8952863503761708,
+        'error_tree_balanced_accuracy': 0.7209989546416461,
         'primary_model_predicted_accuracy': 1, #0.9011208352525718,
```
Comment (AgatheG) [inline mealy/tests/test_error_analyzer.py:115]:
```suggestion
        'primary_model_predicted_accuracy': 1,
```

**Review by AgatheG — CHANGES_REQUESTED** [review]
There are several instances of `Model Performance Predictor` left:
- README.md: l.71, 75, 78, 131, 135, 138
- error_analyzer.py: l.193
- error_visualizer.py: l.77, 105, 209

File: mealy/error_tree.py (hunk 7 above)
Comment (AgatheG) [inline mealy/error_tree.py:87]: Easier to cast one iinteger to float than a whole array
```suggestion
        self._total_error_fraction = wrongly_predicted_samples / float(n_total_errors)

```

**Review by AgatheG — APPROVED** [review]
:+1:

**Review by du-phan — APPROVED** [review]

**Pull request merged**